sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientJavaDocCodeSnippets.java

```diff
@@ -234,6 +235,102 @@ public void recognizeBatchEntitiesMaxOverload() {
+                    + " entity subcategory: %s, offset: %s, length: %s, confidence score: %f.%n",
+                entity.getText(), entity.getCategory(), entity.getSubcategory(), entity.getOffset(),
+                entity.getLength(), entity.getConfidenceScore())));
+        // END: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.recognizePiiEntities#string
+    }
+
+    /**
+     * Code snippet for {@link TextAnalyticsAsyncClient#recognizePiiEntities(String, String)}
+     */
+    public void recognizePiiEntitiesWithLanguage() {
+
+        // BEGIN: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.recognizePiiEntities#string-string
+        String document = "My SSN is 555-55-5555";
+        textAnalyticsAsyncClient.recognizePiiEntities(document, "en")
+            .subscribe(piiEntityCollection -> piiEntityCollection.forEach(entity -> System.out.printf(
+                "Recognized Personally Identifiable Information entity: %s, entity category: %s,"
+                    + " entity subcategory: %s, offset: %s, length: %s, confidence score: %f.%n",
+                entity.getText(), entity.getCategory(), entity.getSubcategory(), entity.getOffset(),
+                entity.getLength(), entity.getConfidenceScore())));
+        // END: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.recognizePiiEntities#string-string
+    }
+
+    /**
+     * Code snippet for {@link TextAnalyticsAsyncClient#recognizePiiEntitiesBatch(Iterable, String, TextAnalyticsRequestOptions)}
+     */
+    public void recognizePiiEntitiesStringListWithOptions() {
+        // BEGIN: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.recognizePiiEntitiesBatch#Iterable-String-TextAnalyticsRequestOptions
+        List<String> documents = Arrays.asList(
+            "My SSN is 555-55-5555.",
+            "Visa card 0111 1111 1111 1111."
+        );
+
+        // Request options: show statistics and model version
+        TextAnalyticsRequestOptions requestOptions = new TextAnalyticsRequestOptions().setIncludeStatistics(true)
+            .setModelVersion("latest");
+
+        textAnalyticsAsyncClient.recognizePiiEntitiesBatch(documents, "en", requestOptions)
+            .subscribe(piiEntitiesResults -> {
+                // Batch statistics
+                TextDocumentBatchStatistics batchStatistics = piiEntitiesResults.getStatistics();
+                System.out.printf("Batch statistics, transaction count: %s, valid document count: %s.%n",
+                    batchStatistics.getTransactionCount(), batchStatistics.getValidDocumentCount());
+
+                piiEntitiesResults.forEach(recognizePiiEntitiesResult ->
+                    recognizePiiEntitiesResult.getEntities().forEach(entity -> System.out.printf(
+                        "Recognized Personally Identifiable Information entity: %s, entity category: %s,"
+                            + " entity subcategory: %s, offset: %s, length: %s, confidence score: %f.%n",
```

Comment:
make sure to keep this concise as this gets added in between the java docs and could make the docs really verbose!